projects/core/components/error/error.component.ts

```diff
@@ -15,13 +15,23 @@ import {Observable} from 'rxjs';
     animations: [tuiHeightCollapse, tuiFadeIn],
 })
 export class TuiErrorComponent {
-    @Input()
+    @Input('error')
     @tuiDefaultProp()
+    set errorSetter(error: TuiValidationError | string | null) {
```

Comment:
@waterplea should we maybe support the whole polymorphous content here?

Reply:
`TuiValidationError` is already a wrapper around polymorpheus + context so I think it's not necessary. What is necessary though is to handle strings as validation errors for controls, not just in `Error` component. Say we have a validator that returns this error: `{ error: 'Oops!' }`.

@podkovyrovda check out `FieldError` component around line 185. At the end of that function we try to get error content from the DI. In case we do not get it, the content is undefined and context is actual string error. We need to handle this case, but also be able to show default error message. I think if we have `content === undefined` and `typeof context === 'string'` there at the end we can use `context` as `content`. This way it would show `'Oops!'` for the validator above and default error message for `{ required: true }` for example. 

@MarsiBarsi what do you think?

P.S. there's also a TODO there, you can remove it and the line it mentions, it is outdated.

Reply:
@MarsiBarsi @podkovyrovda so what do you think?

Reply:
Well, let's do it, a bit mindblowing but solves the case